File: OOP_3A/PartI/Stack.java
class Stack {
    private Node top;

    public Stack() {
        top = null;
    }

    void empty() {
        if (top == null)
            System.out.println("true");
        else
            System.out.println("false");
    }

    void push(int element) {
        if (top == null) {
            Node newNode = new Node(element, null);
            top = newNode;
            System.out.println("new node built");
            return;
        }
        Node newNode = new Node(element, top);
        top = newNode;
        System.out.println("new element insert");
    }

    int pop() {
        int item;
        if (top == null) {
            System.out.println("there is nothing to pop out");
            return Integer.MIN_VALUE;
        }
        Node current = top;
        top = top.next;
        System.out.println("pop out successfully,");
        item = current.item;
        current = null;
        return item;
    }
}

File: OOP_3A/PartI/Stack.cpp
using namespace std;
#include "Node.h"
#include <iostream>
class Stack {
private:
    NodePtr top;
public:
    Stack()
    {
        top = nullptr;
    }
    void empty()
    {
        if(top == nullptr)
            cout<<"true"<<endl;
        else
            cout<<"false"<<endl;
    }
    void  push(int element)
    {
        if(top == nullptr) {
            Node *newNode = new Node(element,nullptr);
            top = newNode;
            cout<<"new node built"<<endl;
            return;
        }
        Node *newNode = new Node(element,top);
        top = newNode;
        cout<<"new element insert"<<endl;
    }
    int pop()
    {
        int item;
        if(top == nullptr)
        {
            cout<<"there is nothing to pop out"<<endl;
            return -100000 ;
        }
        NodePtr current = top;
        top = top->next;
        cout<<"pop out successfully,";
        item = current->item;
        delete current;
        return item;
    }
};
